### DISCUSSION TITLE
Incendiary / molotov grenades

### DISCUSSION THREAD

**Discussion by pedrofornaza** [saul/demofile#214] [Questions] [answered]
We dont have "fire" grenades events. All other grenades have a "detonate" event, but incendiary / molotov doesnt.

Apparently https://github.com/akiver/CSGO-Demos-Manager have it all. Maybe we can parse in a different way, based on them.

**Comment by saul** [comment]
Use the 'inferno_startburn' event

**Comment by pedrofornaza** [comment]
@saul, it can be used for both? How can i identify the "thrower"?

**Comment by saul** [comment]
There's an entityindex event variable that can be passed into `demo.entities.entities[entityindex]`. You can then get the `owner` from this entity to see who threw the grenade. To see if it was a molotov/incendiary, you can check `className` on the weapon entity. Hope that helps

**Comment by pedrofornaza** [comment]
It worked! Thanks.

**Comment by Brammz** [comment]
Does the `className` still work? The following code gives me a position and an owner, but always results in `undefined` for type.

```javascript
demoFile.gameEvents.on('inferno_startburn', e => {
	let grenade = demoFile.entities.entities[e.entityid];
	let type = grenade.className;
	let position = grenade.position;
	let thrower = grenade.owner;
});
```

**Comment by saul** [comment]
Use `grenade.serverClass.name` instead.

**Comment by cAction** [comment]
Does grenade.serverClass.name still work? The name is always the same.

**Comment by saul** [comment, marked as answer]
Sorry for the delay. I have found another method which involves listening (heh) for the explosion sound to be emitted:

```ts
//const SEED = 1146049601;
const molotovExplode = 3580951569; // murmurHash2("inferno.start", SEED);
const incExplode = 1110819271; // murmurHash2("inferno.start_incgrenade", SEED);

demoFile.on("svc_Sounds", e => {
  for (const sound of e.sounds) {
    if (
      sound.soundNumHandle !== molotovExplode &&
      sound.soundNumHandle !== incExplode
    ) {
      continue;
    }

    const inferno = (demoFile.entities.entities.get(
      sound.entityIndex
    ) as unknown) as BaseEntity<CInferno>;
    const pos = inferno.position;

    const thrower = inferno.owner as Player;
    const isMolotov = sound.soundNumHandle === molotovExplode;

    console.log(
      `${isMolotov ? "Molotov" : "Incendiary"} thrown by ${
        thrower.name
      } exploded at (${pos.x},${pos.y},${pos.z})`
    );
  }
});
```

Should print similar to:

```
Molotov thrown by EliGE exploded at (25.3125,2094.28125,-124.71875)
Molotov thrown by bew exploded at (-1539.46875,2723.09375,8.34375)
Incendiary thrown by djay exploded at (-2013.46875,1593.15625,33.375)
Molotov thrown by bew exploded at (550.78125,596.15625,5.15625)
Molotov thrown by EliGE exploded at (-353.96875,1687.75,-127.78125)
```

Hope that helps.

**Comment by cfpperche** [comment]
demoFile.gameEvents.on('inferno_startburn', e => {
	let grenade = demoFile.entities.entities[e.entityid];
	let type = grenade.className;
	let position = grenade.position;
	let thrower = grenade.owner;
});


granade is as undefined in all events

**Reply by saul** [reply]
See the accepted answer: https://github.com/saul/demofile/discussions/214#discussioncomment-414808

I've also just added a molotov example to the repo: https://github.com/saul/demofile/blob/master/examples/molotov.ts
